src/main/java/bio/terra/cloudfunctions/common/CloudEventsHarness.java

```diff
@@ -0,0 +1,99 @@
+package bio.terra.cloudfunctions.common;
+
+import bio.terra.cloudevents.v1.CloudEventType;
+import bio.terra.cloudevents.v1.messagewrapper.FirestoreEventMessage;
+import bio.terra.cloudevents.v1.messagewrapper.PubSubEventMessage;
+import bio.terra.cloudevents.v1.messagewrapper.StorageObjectEventMessage;
+import com.google.cloud.functions.CloudEventsFunction;
+import com.google.events.cloud.firestore.v1.DocumentEventData;
+import com.google.events.cloud.pubsub.v1.MessagePublishedData;
+import com.google.events.cloud.storage.v1.StorageObjectData;
+import io.cloudevents.CloudEvent;
+import java.util.logging.Logger;
+
+/**
+ * This class is an abstract representation of a Cloud Function and contains only framework-specific
+ * logic to receive and parse CloudEvent messages.
+ *
+ * <p>Sub-classes of this class can be integrated with a DI framework to deploy Cloud Functions like
+ * a Service.
+ */
+public abstract class CloudEventsHarness implements CloudEventsFunction {
+  private static final Logger logger = Logger.getLogger(CloudEventsHarness.class.getName());
+
+  private CloudEvent event;
+  private Object message;
+
+  /**
+   * Subclasses of this class must override and invoke this method at the minimum.
+   *
+   * @param event
+   * @throws Exception
+   */
+  @Override
+  public void accept(CloudEvent event) throws Exception {
+    this.event = event;
+    parse();
+  }
+
+  public CloudEvent getEvent() {
+    return event;
+  }
+
+  public Object getMessage() {
+    return message;
+  }
+
+  public CloudEventType getCloudEventType() {
+    return CloudEventType.fromCode(event.getType());
+  }
+
+  /**
+   * The parse() method delegates message parsing to specific implementation that can be replaced in
+   * the fture.
+   *
+   * @throws Exception
+   */
+  public void parse() throws Exception {
+    parseCloudEventV1();
+  }
+
+  private void parseCloudEventV1() throws Exception {
```

Comment:
Could you shed some light on why we would use `parseCloudEventV1` at one point, and `parseCloudEventV03` other times? Is there possibly a way the reason could be communicated using the function names themselves?

Reply:
The Google `cloudevents` API has a specification for V03 Cloud Event and I added `parseCloudEventV03` as a placeholder. I have removed this method as it is unused. In the future, we can encapsulate version logic in its own method.